rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/retrieval/TrustAnchorRetrievalService.java

```diff
@@ -0,0 +1,147 @@
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.http.HttpHeader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+@Slf4j
+@Setter
+@Service
+public class TrustAnchorRetrievalService {
+    private boolean isFileProtocolEnabled = false;
+
+    @Value("${rpki.validator.rsync.local.storage.directory}")
+    private File localRsyncStorageDirectory;
+
+    @Autowired
+    private BuildInformation buildInformation;
+
+    @Autowired
+    private RsyncFactory rsyncFactory;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    @Autowired
+    private HttpClientMetricsService httpMetrics;
+
+    @Autowired
+    private RsyncMetricsService rsyncMetrics;
+
+    public byte[] fetchTrustAnchorCertificate(URI trustAnchorCertificateURI, ValidationResult validationResult) throws IOException {
+        switch (trustAnchorCertificateURI.getScheme()) {
+            case "rsync":
+                return fetchRsyncTrustAnchorCertificate(trustAnchorCertificateURI, validationResult);
+            case "https":
+                return fetchHttpsTrustAnchorCertificate(trustAnchorCertificateURI, validationResult);
+            case "file":
+                if (isFileProtocolEnabled) {
+                    return Files.toByteArray(new File(trustAnchorCertificateURI));
+                }
+                validationResult.error(ErrorCodes.TRUST_ANCHOR_FETCH, trustAnchorCertificateURI.toASCIIString(), "File support not explicitly enabled.");
+                return null;
+            default:
+                validationResult.error(ErrorCodes.TRUST_ANCHOR_FETCH, trustAnchorCertificateURI.toASCIIString(), "Unsupported URI");
+                return null;
+        }
+    }
+
+    protected byte[] fetchHttpsTrustAnchorCertificate(URI trustAnchorCertificateURI, ValidationResult validationResult) throws IOException {
+        final long t0 = System.currentTimeMillis();
+        String statusDescription = "unknown";
+
+        try {
+            ContentResponse res = httpClient
+                    .newRequest(trustAnchorCertificateURI)
+                    .header(HttpHeader.USER_AGENT, null)
+                    .header(HttpHeader.USER_AGENT, String.format("RIPE NCC RPKI Validator/%s", buildInformation.getVersion()))
+                    .send();
+            log.debug("HTTP {} when fetching HTTPS trust anchor from {}", res.getStatus(), trustAnchorCertificateURI);
+            statusDescription = String.valueOf(res.getStatus());
+            if (res.getStatus() != 200) {
+                validationResult.error(ErrorCodes.TRUST_ANCHOR_FETCH, trustAnchorCertificateURI.toASCIIString(), String.format("HTTP %d", res.getStatus()));
```

Comment:
nit: Instead of formatting status code into "HTTP %d", maybe supply res.getReason().